datafusion/functions/src/datetime/make_date.rs

```diff
@@ -122,6 +122,12 @@ impl ScalarUDFImpl for MakeDateFunc {
 
         let [years, months, days] = take_function_args(self.name(), args)?;
 
+        if matches!(years, ColumnarValue::Scalar(ScalarValue::Null)) 
+            || matches!(months, ColumnarValue::Scalar(ScalarValue::Null))
+            || matches!(days, ColumnarValue::Scalar(ScalarValue::Null)) {
+            return Ok(ColumnarValue::Scalar(ScalarValue::Null));
+        }
```

Comment:
nit:
```suggestion
        if [years, months, days].iter().any(|v| matches!(v, ColumnarValue::Scalar(ScalarValue::Null))) {
            return Ok(ColumnarValue::Scalar(ScalarValue::Null));
        }
```

Reply:
Suggested change when applied, compiler asks me to convert it to something like this:
```
    if [years.clone(), months.clone(), days.clone()].iter().any(|v| matches!(v, ColumnarValue::Scalar(ScalarValue::Null))) {
            return Ok(ColumnarValue::Scalar(ScalarValue::Null));
	}
```

this will `clone` the Scalar values, is that acceptable?